File: bms/go/geo/geo_s1_ChordAngle_Angle/geo_s1_ChordAngle_Angle.go
package test

import (
	"math"
)

const (
	geo_s1_Radian geo_s1_Angle = 1
)

func (c geo_s1_ChordAngle) Angle() geo_s1_Angle {
	if c < 0 {
		return -1 * geo_s1_Radian
	}
	if c.IsInfinity() {
		return geo_s1_InfAngle()
	}
	return geo_s1_Angle(2 * math.Asin(0.5*math.Sqrt(float64(c))))
}

func (c geo_s1_ChordAngle) IsInfinity() bool {
	return math.IsInf(float64(c), 1)
}

func geo_s1_InfAngle() geo_s1_Angle {
	return geo_s1_Angle(math.Inf(1))
}

type geo_s1_ChordAngle float64

type geo_s1_Angle float64

File: translations_only/fail/go/geo/geo_s1_ChordAngle_Angle/claude2/Restart/geo_s1_ChordAngle_Angle.rs

use std::f64::INFINITY;

struct GeoS1Angle(f64);

impl GeoS1Angle {
    fn new(angle: f64) -> GeoS1Angle {
        GeoS1Angle(angle)
    }
}

fn geo_s1_radian() -> GeoS1Angle {
    GeoS1Angle::new(1.0)  
}

struct GeoS1ChordAngle(f64);

impl GeoS1ChordAngle {

    fn angle(&self) -> GeoS1Angle {
        if self.0 < 0.0 {
            return GeoS1Angle::new(-1.0 * geo_s1_radian().0);
        }
        if self.is_infinity() {
            return geo_s1_inf_angle();
        }
        GeoS1Angle::new(2.0 * (0.5 * self.0).asin())
    }

    fn is_infinity(&self) -> bool {
        self.0.is_infinite()
    }
}

fn geo_s1_inf_angle() -> GeoS1Angle {
    GeoS1Angle::new(INFINITY)
}

